### PR TITLE
Simplify linear interpolation in forecast.py

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/model/forecast.py [hunk 1]
```diff
@@ -24,67 +24,27 @@
 from gluonts.core.component import validated
 
 
-class LinearInterpolation:
-    """
-    Linear interpolation based on datapoints (x_coord, y_coord)
+def _linear_interpolation(
+    xs: np.ndarray, ys: np.ndarray, x: float
+) -> np.ndarray:
+    assert sorted(xs) == xs
```
Comment (melopeo) [inline src/gluonts/model/forecast.py:30]: Can we complement the asserts with a text message?

Comment (jaheba) [reply src/gluonts/model/forecast.py:30]: We could, but I don't see much value in it. The traceback should already contain enough information. Also, this function is only used internally.

Comment (melopeo) [reply src/gluonts/model/forecast.py:30]: Sounds good.

**Review by melopeo — APPROVED** [review]
LGTM

**Pull request merged**